# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1351921908 -3600
#      Sat Nov 03 06:51:48 2012 +0100
# Node ID ae0af6c664c48030d4c863b57adbec52121b4ba2
# Parent  0754bdfbc8fecbbf3b6111a8ee04562896b43815
backout changeset 5649e84ea3ce (see discussion on patch #7867)

diff --git a/libgui/src/m-editor/file-editor-interface.h b/libgui/src/m-editor/file-editor-interface.h
--- a/libgui/src/m-editor/file-editor-interface.h
+++ b/libgui/src/m-editor/file-editor-interface.h
@@ -39,26 +39,16 @@ class file_editor_interface : public QDo
     : QDockWidget ((QWidget*)mainWindow) // QDockWidget constructor is explicit, hence the cast.
   {
     setObjectName ("FileEditor");
     _terminal = terminal;
     _main_window = mainWindow;
 
     connect (this, SIGNAL (visibilityChanged (bool)), this,
              SLOT (handle_visibility_changed (bool)));
-
-    // connect the editor dockWidget's topLevelChanged signal, which
-    // is emitted when its floating property changes, to a
-    // user-defined slot. We do this so than when the editor is popped
-    // out we can turn it into a proper window with all the normal
-    // toolbar buttons for a window, i.e. minimise, maximise, close
-    connect (this,
-             SIGNAL(topLevelChanged(bool)), this,
-             SLOT(dockWidget_topLevelChanged(bool)));
-
   }
 
   virtual ~file_editor_interface () { }
 
   virtual QMenu *debug_menu () = 0;
   virtual QToolBar *toolbar () = 0;
 
   virtual void handle_entered_debug_mode () = 0;
@@ -83,32 +73,11 @@ protected:
   }
 
 protected slots:
   void handle_visibility_changed (bool visible)
   {
     if (visible)
       emit active_changed (true);
   }
-
-  // when the floating property of a dockWidget is changed from docked
-  // to floating we make it a top level window (with minmize,
-  // maximize, and close button in the title bar) by calling
-  // setWindowFlags(Qt::Window)
-  //
-  // The dockWidget will automatically regain it's Qt::widget flag
-  // when it becomes docked again (by dragging it to the right place
-  // or double clicking the title bar)
-  void dockWidget_topLevelChanged (bool isFloating)
-  {
-      if (isFloating)
-        {
-          //file_editor.setWindowFlags(Qt::Window);
-          this->setWindowFlags(Qt::Window);
-          // setWindowFlags calls setParent() when changing the flags
-          // for a window, causing the widget to be hidden. We must
-          // call show() to make the widget visible again
-          this->show();
-        }
-  }
 };
 
 #endif // FILEEDITORINTERFACE_H
